Review of reported errors from local run CAMARA linter to confirm conformance with API Design Guidelines for Traffic_Influence.yaml

After running locally CAMARA linter scripts indicated in https://github.com/camaraproject/EdgeCloud/discussions/170, although using [draft spectral](https://github.com/camaraproject/Commonalities/blob/aaeb455e3400347de1ae6558337ef739cb943837/artifacts/linting_rules/.spectral.yml) , along with files in [lint_function_folder](https://github.com/camaraproject/Commonalities/tree/aaeb455e3400347de1ae6558337ef739cb943837/artifacts/linting_rules/lint_function), following errors/warnings appear:

```
   5:6   warning  info-contact                          Info object must have "contact" object.                                                                                                          info
  124:7    error  oas3-valid-schema-example             "example" property must match format "ipv4"                                                                                                      paths
125:23    error  camara-parameter-casing-convention    /traffic-influences should be kebab-case: must be kebab case                                                                                     paths./traffic-influences
134:11  warning  operation-tag-defined                 Operation tags must be defined in global tags.                                                                                                   paths./traffic-influences.get.tags[0]
137:20     hint  camara-operationid-casing-convention  Operation Id must be in Camel case "must be camel case"                                                                                          paths./traffic-influences.get.operationId
160:11  warning  operation-tag-defined                 Operation tags must be defined in global tags.                                                                                                   paths./traffic-influences.post.tags[0]
164:20     hint  camara-operationid-casing-convention  Operation Id must be in Camel case "must be camel case"                                                                                          paths./traffic-influences.post.operationId
201:44    error  camara-parameter-casing-convention    /traffic-influences/{trafficInfluenceID} should be kebab-case: must be kebab case                                                                paths./traffic-influences/{trafficInfluenceID}
202:16    error  camara-http-methods                   Invalid HTTP method for '#/paths/~1traffic-influences~1{trafficInfluenceID}/parameters'. Must be one of get, put, post, delete, patch, options.  paths./traffic-influences/{trafficInfluenceID}.parameters
  210:9  warning  operation-description                 Operation "description" must be present and non-empty string.                                                                                    paths./traffic-influences/{trafficInfluenceID}.get
  210:9  warning  operation-operationId                 Operation must have "operationId".                                                                                                               paths./traffic-influences/{trafficInfluenceID}.get
213:11  warning  operation-tag-defined                 Operation tags must be defined in global tags.                                                                                                   paths./traffic-influences/{trafficInfluenceID}.get.tags[0]
235:11  warning  operation-tag-defined                 Operation tags must be defined in global tags.                                                                                                   paths./traffic-influences/{trafficInfluenceID}.patch.tags[0]
238:20     hint  camara-operationid-casing-convention  Operation Id must be in Camel case "must be camel case"                                                                                          paths./traffic-influences/{trafficInfluenceID}.patch.operationId
276:12  warning  operation-description                 Operation "description" must be present and non-empty string.                                                                                    paths./traffic-influences/{trafficInfluenceID}.delete
276:12  warning  operation-operationId                 Operation must have "operationId".                                                                                                               paths./traffic-influences/{trafficInfluenceID}.delete
279:11  warning  operation-tag-defined                 Operation tags must be defined in global tags.                                                                                                   paths./traffic-influences/{trafficInfluenceID}.delete.tags[0]
324:31  warning  camara-parameters-descriptions        Parameter description is missing or empty: "onTrafficInfluenceChanged.description" property must be truthy                                       components.callbacks.onTrafficInfluenceChanged
324:31    error  oas3-valid-schema-example             "example" property must match format "ipv4"                                                                                                      components.callbacks.onTrafficInfluenceChanged
350:22  warning  camara-parameters-descriptions        Parameter description is missing or empty: "TrafficInfluence.description" property must be truthy                                                components.schemas.TrafficInfluence
350:22  warning  camara-parameters-descriptions        Parameter description is missing or empty: "TrafficInfluence" property must be truthy                                                            components.schemas.TrafficInfluence
402:27  warning  camara-parameters-descriptions        Parameter description is missing or empty: "PatchTrafficInfluence.description" property must be truthy                                           components.schemas.PatchTrafficInfluence
406:28  warning  camara-parameters-descriptions        Parameter description is missing or empty: "trafficInfluenceID.description" property must be truthy                                              components.schemas.PatchTrafficInfluence.properties.trafficInfluenceID
408:23  warning  camara-parameters-descriptions        Parameter description is missing or empty: "apiConsumerId.description" property must be truthy                                                   components.schemas.PatchTrafficInfluence.properties.apiConsumerId
410:23  warning  camara-parameters-descriptions        Parameter description is missing or empty: "applicationId.description" property must be truthy                                                   components.schemas.PatchTrafficInfluence.properties.applicationId
412:15  warning  camara-parameters-descriptions        Parameter description is missing or empty: "state.description" property must be truthy                                                           components.schemas.PatchTrafficInfluence.properties.state
415:26  warning  camara-parameters-descriptions        Parameter description is missing or empty: "PostTrafficInfluence.description" property must be truthy                                            components.schemas.PostTrafficInfluence
419:28  warning  camara-parameters-descriptions        Parameter description is missing or empty: "trafficInfluenceID.description" property must be truthy                                              components.schemas.PostTrafficInfluence.properties.trafficInfluenceID
421:15  warning  camara-parameters-descriptions        Parameter description is missing or empty: "state.description" property must be truthy                                                           components.schemas.PostTrafficInfluence.properties.state
424:34  warning  camara-parameters-descriptions        Parameter description is missing or empty: "TrafficInfluenceNotification.description" property must be truthy                                    components.schemas.TrafficInfluenceNotification
462:29  warning  camara-parameters-descriptions        Parameter description is missing or empty: "NetworkAccessIdentifier" property must be truthy                                                     components.schemas.NetworkAccessIdentifier
462:29  warning  camara-parameters-descriptions        Parameter description is missing or empty: "NetworkAccessIdentifier.description" property must be truthy                                         components.schemas.NetworkAccessIdentifier
476:16    error  oas3-valid-schema-example             "example" property must match format "ipv4"                                                                                                      components.schemas.Ipv4Address.example
509:17  warning  camara-parameters-descriptions        Parameter description is missing or empty: "ErrResponse.description" property must be truthy                                                     components.schemas.ErrResponse
512:16  warning  camara-parameters-descriptions        Parameter description is missing or empty: "status.description" property must be truthy                                                          components.schemas.ErrResponse.properties.status
515:17  warning  camara-parameters-descriptions        Parameter description is missing or empty: "message.description" property must be truthy                                                         components.schemas.ErrResponse.properties.message
518:15  warning  camara-parameters-descriptions        Parameter description is missing or empty: "ErrorInfo.description" property must be truthy                                                       components.schemas.ErrorInfo
592:24  warning  oas3-unused-component                 Potentially unused component has been detected.                                                                                                  components.responses.SessionNotFound404
```


To be double checked if correction is needed in yaml to avoid those errors/warnings ( at least the kebab case one seems a false positive). Most of them are warnings for missing parameter descriptions.

Hi Fabrizio, apparently draft version had a bug checking Kebab-case in path. Links for linter files from commonalities (referenced in in https://github.com/camaraproject/EdgeCloud/discussions/170)  are working now (updated in main branch). We have rerun it with current Traffic_Influence.yaml ( the one in progress in https://github.com/camaraproject/EdgeCloud/pull/167

And current output is the following:

```
   5:6   warning  info-contact                          Info object must have "contact" object.                                                                                                          info
  123:7    error  oas3-valid-schema-example             "example" property must match format "ipv4"                                                                                                      paths
133:11  warning  operation-tag-defined                 Operation tags must be defined in global tags.                                                                                                   paths./traffic-influences.get.tags[0]
136:20     hint  camara-operationid-casing-convention  Operation Id must be in Camel case "must be camel case"                                                                                          paths./traffic-influences.get.operationId
159:11  warning  operation-tag-defined                 Operation tags must be defined in global tags.                                                                                                   paths./traffic-influences.post.tags[0]
162:20     hint  camara-operationid-casing-convention  Operation Id must be in Camel case "must be camel case"                                                                                          paths./traffic-influences.post.operationId
200:16    error  camara-http-methods                   Invalid HTTP method for '#/paths/~1traffic-influences~1{trafficInfluenceID}/parameters'. Must be one of get, put, post, delete, patch, options.  paths./traffic-influences/{trafficInfluenceID}.parameters
  208:9  warning  camara-routes-description             Functionality method description Warning: Each method should have description.                                                                   paths./traffic-influences/{trafficInfluenceID}.get
  208:9  warning  operation-description                 Operation "description" must be present and non-empty string.                                                                                    paths./traffic-influences/{trafficInfluenceID}.get
  208:9  warning  operation-operationId                 Operation must have "operationId".                                                                                                               paths./traffic-influences/{trafficInfluenceID}.get
211:11  warning  operation-tag-defined                 Operation tags must be defined in global tags.                                                                                                   paths./traffic-influences/{trafficInfluenceID}.get.tags[0]
233:11  warning  operation-tag-defined                 Operation tags must be defined in global tags.                                                                                                   paths./traffic-influences/{trafficInfluenceID}.patch.tags[0]
236:20     hint  camara-operationid-casing-convention  Operation Id must be in Camel case "must be camel case"                                                                                          paths./traffic-influences/{trafficInfluenceID}.patch.operationId
274:12  warning  camara-routes-description             Functionality method description Warning: Each method should have description.                                                                   paths./traffic-influences/{trafficInfluenceID}.delete
274:12  warning  operation-description                 Operation "description" must be present and non-empty string.                                                                                    paths./traffic-influences/{trafficInfluenceID}.delete
274:12  warning  operation-operationId                 Operation must have "operationId".                                                                                                               paths./traffic-influences/{trafficInfluenceID}.delete
277:11  warning  operation-tag-defined                 Operation tags must be defined in global tags.                                                                                                   paths./traffic-influences/{trafficInfluenceID}.delete.tags[0]
322:31  warning  camara-properties-descriptions        Property description is missing or empty: "onTrafficInfluenceChanged.description" property must be truthy                                        components.callbacks.onTrafficInfluenceChanged
322:31    error  oas3-valid-schema-example             "example" property must match format "ipv4"                                                                                                      components.callbacks.onTrafficInfluenceChanged
348:22  warning  camara-properties-descriptions        Property description is missing or empty: "TrafficInfluence.description" property must be truthy                                                 components.schemas.TrafficInfluence
348:22  warning  camara-properties-descriptions        Property description is missing or empty: "TrafficInfluence" property must be truthy                                                             components.schemas.TrafficInfluence
398:27  warning  camara-properties-descriptions        Property description is missing or empty: "PatchTrafficInfluence.description" property must be truthy                                            components.schemas.PatchTrafficInfluence
402:28  warning  camara-properties-descriptions        Property description is missing or empty: "trafficInfluenceID.description" property must be truthy                                               components.schemas.PatchTrafficInfluence.properties.trafficInfluenceID
404:23  warning  camara-properties-descriptions        Property description is missing or empty: "apiConsumerId.description" property must be truthy                                                    components.schemas.PatchTrafficInfluence.properties.apiConsumerId
406:15  warning  camara-properties-descriptions        Property description is missing or empty: "appId.description" property must be truthy                                                            components.schemas.PatchTrafficInfluence.properties.appId
408:15  warning  camara-properties-descriptions        Property description is missing or empty: "state.description" property must be truthy                                                            components.schemas.PatchTrafficInfluence.properties.state
411:26  warning  camara-properties-descriptions        Property description is missing or empty: "PostTrafficInfluence.description" property must be truthy                                             components.schemas.PostTrafficInfluence
415:28  warning  camara-properties-descriptions        Property description is missing or empty: "trafficInfluenceID.description" property must be truthy                                               components.schemas.PostTrafficInfluence.properties.trafficInfluenceID
417:15  warning  camara-properties-descriptions        Property description is missing or empty: "state.description" property must be truthy                                                            components.schemas.PostTrafficInfluence.properties.state
420:34  warning  camara-properties-descriptions        Property description is missing or empty: "TrafficInfluenceNotification.description" property must be truthy                                     components.schemas.TrafficInfluenceNotification
458:29  warning  camara-properties-descriptions        Property description is missing or empty: "NetworkAccessIdentifier" property must be truthy                                                      components.schemas.NetworkAccessIdentifier
458:29  warning  camara-properties-descriptions        Property description is missing or empty: "NetworkAccessIdentifier.description" property must be truthy                                          components.schemas.NetworkAccessIdentifier
472:16    error  oas3-valid-schema-example             "example" property must match format "ipv4"                                                                                                      components.schemas.Ipv4Address.example
510:17  warning  camara-properties-descriptions        Property description is missing or empty: "ErrResponse.description" property must be truthy                                                      components.schemas.ErrResponse
513:16  warning  camara-properties-descriptions        Property description is missing or empty: "status.description" property must be truthy                                                           components.schemas.ErrResponse.properties.status
516:17  warning  camara-properties-descriptions        Property description is missing or empty: "message.description" property must be truthy                                                          components.schemas.ErrResponse.properties.message
519:15  warning  camara-properties-descriptions        Property description is missing or empty: "ErrorInfo.description" property must be truthy                                                        components.schemas.ErrorInfo
593:24  warning  oas3-unused-component                 Potentially unused component has been detected.                                                                                                  components.responses.SessionNotFound404
```

--- a/.spectral.yml
+++ b/.spectral.yml
@@ -1,6 +1,9 @@
 # CAMARA Project - linting ruleset - documentation avaialable here:
 # https://github.com/camaraproject/Commonalities/blob/main/documentation/Linting-rules.md
-# 31.01.2024 - initial version
+# Changelog:
+# - 31.01.2024: Initial version
+# - 19.03.2024: Corrected camara-http-methods rule
+
 
 extends: "spectral:oas"
 functions:
@@ -108,7 +111,7 @@ rules:
     then:
       function: pattern
       functionOptions:
-        match: "^(get|put|post|delete|patch|options)$"
+        match: "^(get|put|post|delete|patch|options|parameters)$"
     recommended: true  # Set to true/false to enable/disable this rule
 
   camara-get-no-request-body:

